### PR TITLE
Error if users attempt to do `tsh login --headless`

### PR DESCRIPTION
**Pull request by codingllama**
Instead of silently defaulting to local login, which can run into other errors and potentially confuse the user (it did confuse us), we now error if the user attempts to do `tsh login --headless`.

I've snuck in a fix for the "SSO login warning" logic, as it was showing for me on a local cluster.

#28045

### CODE REVIEW

File: lib/client/api.go [hunk 1]
```diff
@@ -3531,8 +3531,9 @@ func (tc *TeleportClient) getSSHLoginFunc(pr *webclient.PingResponse) (SSHLoginF
 					return tc.headlessLogin(ctx, priv)
 				}, nil
 			}
-			log.Debug("Headless login is disabled for this command. Only 'tsh ls', 'tsh ssh', and 'tsh scp' are supported. Defaulting to local authentication methods.")
-			fallthrough
+			return nil, trace.BadParameter("" +
+				"Headless login is disabled for this command. " +
```
Comment (zmb3) [inline lib/client/api.go:3535]:
```suggestion
				"Headless login is not supported for this command. " +
```

**Review by zmb3 — APPROVED** [review]
Minor suggestion, take it or leave it

**Review by Joerger — APPROVED** [review]

File: lib/client/api.go (hunk 1 above)
Comment (codingllama) [reply lib/client/api.go]: Sounds better, thanks!

**Comment by codingllama** [comment]
Thanks for the quick review, folks!

**Pull request merged**